node-webservice/react-client/src/Components/HomePage.jsx

```diff
@@ -7,7 +7,13 @@ const HomePage = props => (
         <h1>Join a game</h1>
         <JoinGameForm {...props}/>
         <br/>
-        <Link to="/host">Host a game</Link><br/>
+        <form>
+            <button type="submit" formaction="/host" 
```

Comment:
this formAction does a form submit, causing a page reload and adding the query ? to the url.
<img width="985" alt="Screen Shot 2020-04-01 at 11 41 14 AM" src="https://user-images.githubusercontent.com/11395364/78174274-d7ac1f80-740d-11ea-8fe1-a84dbac2e053.png">

The `<Link>` tag is preferable here because it will call into React router to do a dynamic page change, so no page reload is needed. The page reload is probably very fast on a dev environment, but on production it will be slower. 

I suggest switching this to be a Button that uses the renderAs prop so that it can be both a `<button>` and also a react router `<Link>` https://github.com/couds/react-bulma-components/issues/55#issuecomment-400172397